tests/ndarrays/conftest.py

```diff
@@ -0,0 +1,118 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=arguments-differ, inconsistent-return-statements, protected-access, abstract-method/
+import subprocess
+import os
+import sys
+import pytest
+
+def pytest_collect_file(parent, path):
+    """
+    A hook to collect test_*.c test files.
+
+    """
+    if path.ext == ".c" and path.basename.startswith("test_"):
+        return CTestFile.from_parent(parent, path)
+
+def pytest_collection_modifyitems(items):
+    """
+    a hook to modify the items before the tests for C unit test files.
+
+    """
+    for item in items:
+        if item.fspath.ext == ".c":
+            item._nodeid = item.nodeid + " < " + item.test_result["DSCR"] + " >"
+
+class CTestFile(pytest.File):
+    """
+    A custom file handler class for C unit test files.
+
+    """
+    @classmethod
+    def from_parent(cls, parent, fspath):
+        return super().from_parent(parent=parent, fspath=fspath)
+
+    def collect(self):
+        """
+        Overridden collect method to collect the results from each
+        C unit test executable.
+
+        """
+        # Run the exe that corresponds to the .c file and capture the output.
+        test_exe = os.path.splitext(str(self.fspath))[0]
+        rootdir = str(self.config.rootdir)
+        test_exe = os.path.relpath(test_exe)
+        if  sys.platform.startswith("win"):
+            ndarray_path =  rootdir + "\\pyccel\\stdlib\\ndarrays\\"
+        else :
+            ndarray_path = rootdir + "/pyccel/stdlib/ndarrays/"
+        comp_cmd = "gcc "+test_exe+".c "+ndarray_path+"*.c -I "+ndarray_path+" -o "+test_exe
+        subprocess.run(comp_cmd, check= 'TRUE' ,shell = 'TRUE')

```

Comment:
When possible it is better to not use `shell = True` (see issue #517 ). To avoid this problem you need to pass a list instead of a string for the command. Something like:
```
comp_cmd = [shutil.which("gcc"), self.fspath, os.path.join(ndarray_path,"*.c"), "-I", ndarray_path, "-o", test_exe]
```